src/poly/mod.rs

```diff
@@ -61,27 +111,76 @@ impl<F, V, M> Expr<F, V, M> {
             Expr::MI(_, m) => m,
         }
     }
-}
 
-#[derive(Debug, Clone)]
-pub struct HashResult {
-    pub hash: u64,
-    pub degree: usize,
+    pub fn map_meta<N: Meta, Func: Fn(&M) -> N>(&self, f: Func) -> Expr<F, V, N> {
```

Comment:
@rutefig why do you want the map to work only if M is of type Meta, why limit it to this?